Runtime/Managers/EditMediator.cs

```diff
@@ -271,36 +276,66 @@ private async Task LoadObjectsState(JObject requestPayload)
                         TypeHelpers.NotNull(objectState["material"]["texture"]);
                 })
                 .GroupBy(objectState => objectState["material"]["id"].ToObject<int>());
-            await ApplyMaterialsToSceneObjects(objectStatesByMaterial);
+            TaskResult materialWasChanged = await ApplyMaterialsToSceneObjects(objectStatesByMaterial);
 
+            TaskResult finalResult = TaskResult.CombineResults(materialWasChanged, colorWasChanged);
+            if (!finalResult.IsSuccess)
+            {
+                originalSceneController?.RestoreOriginalScene();
+                SendFailureLoadSceneMessage(requestPayload["requestTimestamp"], finalResult.Error);
+                return;
+            }
+
+            SendSuccessLoadSceneMessage(requestPayload["requestTimestamp"]);
+        }
+
+        private void SendSuccessLoadSceneMessage(JToken requestPayload)
```

Comment:
```suggestion
        private void SendSuccessLoadSceneMessage(JToken requestTimestamp)
```
This is no longer the same as `requestPayload` in the calling function. It's confusing to have identically named local variables that don't actually represent the same thing. I'd probably pass in the original `requestPayload` object just for clarity. Or pass in a string representing the timestamp so we have a more strongly typed parameter.